nixos/modules/services/web-apps/galene.nix

```diff
@@ -0,0 +1,187 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.galene;
+  defaultRecordingsRootPath = "${cfg.stateDir}/recordings";
+  defaultGroupsRootPath = "${cfg.stateDir}/groups";
+  defaultDataRootPath = "${cfg.stateDir}/data";
+in
+{
+  options = {
+    services.galene = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = "Enable Galene Service.";
+      };
+
+      stateDir = mkOption {
```

Comment:
And do we actually need a `stateDir` option given all the others directory options? Maybe we can just hardcode `/var/lib/galene` because the other options exist.

Reply:
I feel it's good to let people decide where they want to put their state, and even if the other options exist, they default to reside in  ```${cfg.stateDir}````which means you can change a single parameter (the stateDir) without having to change all of the other options.

Reply:
Sorry, what I specifically meant is... are any files stored directly in `cfg.stateDir`, or are all files stored under the other directories specified by `-data`, `-groups`, `-recordings`, and `-static`? After looking at the upstream project for a few short minutes it looked like `cfg.stateDir` doesn't actually hold any files... it just happened to have the other specified directories in it.

I didn't test this, though... so please let me know if I'm wrong.

Reply:
No, no files are stored directly in ```cfg.stateDir``` but the other directories.

Reply:
If the only directories stored in `cfg.stateDir` are already specified via other options then `stateDir` as an option is redundant... right? I guess it just means more typing because less defaults... 

OK, take your pick. Either way works equally fine, just non technical irrelevant opinion on my part at this point I guess :smile: